### PR TITLE
Appended file extension instead of replacing

### PR DESCRIPTION
**Pull request by KingOfBerries**
Changed the path of the asset metafile to `assetname.extension.dclasset` from `assetname.dclasset`.
#861mczpda

### CODE REVIEW

**Comment by KingOfBerries** [comment]
I wanted to use Path.Join but that requires importing a library...

**Comment by MGH-Automator** [comment]
Task linked: [CU-861mczpda .dclasset files should contain the original extension of the asset](https://app.clickup.com/t/861mczpda)

**Review by cblech — CHANGES_REQUESTED** [review]
This does not work for "old" assets. Please write a FileUpgrader for that.
If you need help with that, feel free to contact me

File: Assets/Scripts/System/FileAssetLoaderSystem.cs [hunk 1]
```diff
@@ -178,6 +180,8 @@ public bool SetThumbnailById(Guid id, Texture2D newThumbnail)
         #region Metadata related methods
         private AssetHierarchyItem ScanDirectory(string fileSystemPath, string pathInHierarchy = "")
         {
+            CheckUpgrades(Directory.GetFiles(fileSystemPath, "*.*"));
```
Comment (cblech) [inline Assets/Scripts/System/FileAssetLoaderSystem.cs]: You should be able to set the filter to `"*.dclasset"`. That way, you can remove the test for `.dclasset` in [line 225](#diff-3264eb4ef2e37c77f1ddc226dc3a48110b11b3e6b172dcbdd7305b5a3f9b3a9aR225)

**Review by cblech — CHANGES_REQUESTED** [review]

File: Assets/Scripts/System/FileAssetLoaderSystem.cs (hunk 1 above)
Comment (KingOfBerries) [reply Assets/Scripts/System/FileAssetLoaderSystem.cs]: true

**Review by cblech — APPROVED** [review]

**Pull request merged**